pkgs/development/tools/build-managers/bazel/default.nix

```diff
@@ -190,7 +192,7 @@ stdenv.mkDerivation rec {
   installPhase = ''
     mkdir -p $out/bin
     mv output/bazel $out/bin
-    wrapProgram "$out/bin/bazel" --set JAVA_HOME "${jdk}"
+    wrapProgram "$out/bin/bazel" --set JAVA_HOME "${runJdk}"
```

Comment:
I wonder if you should use `--set-default` here? Then every user could also just set `JAVA_HOME` instead of having to compile there own bazel version?

Reply:
JAVA_HOME is also set by the nix-shell itself:

```
$ nix-shell -p jdk --command 'echo $JAVA_HOME'
/nix/store/6ickqcxhghnlabk0hfzkmlz8za945dk3-openjdk-8u181b13/lib/openjdk
```

Reply:
I see a lot of other derivations using this pattern, I think it's intentional to make them use the Nix jdk / jre instead of the system installed one: `JAVA_HOME` tends to be set in many users' shell and they expect Nix packages to use the dependencies from Nix and not from their system.